application/models/post.go

```diff
@@ -759,8 +759,16 @@ func (p *Post) FindByUserAndUUID(ctx context.Context, user User, uuid string) er
 		Where("uuid = ?", uuid).First(p)
 }
 
-// FindByUser finds all posts visible to the current user.
-func (p *Posts) FindByUser(ctx context.Context, user User, destination, origin *Location, searchText *string) error {
+// PostFilterParams are optional parameters to narrow the list of posts returned from a query
+type PostFilterParams struct {
```

Comment:
Good addition.

Reply:
The addition of the post ID was @fillup 's idea. I started by searching through all visible posts in a Go loop.